### PR TITLE
Fixes #26219 - state control from Rec Logic

### PR DESCRIPTION
**Pull request by sjha4**
****
Today, a user can enable and disable a sync plan from the sync plan page/API and it will enable/diable the associated foreman tasks recurring logic.

This PR adds the ability to enable/disable the foreman task recurring logic directly and tie that change back to the sync plan and maintain consistency between the 2 states.
****
#### **Some use cases:**
1. Create a sync plan.
2. Open the associated recurring logic
3. Disable the recurring logic
4. Check the sync plan enabled status. This should now be false. (UI needs a refresh)
***
1. For the above use-case, go to the recurring logic you disabled. Enable it.
2. The sync plan should also be enabled now.
***
1. Cancel the above recurring logic
2. Sync plan should be in a disabled state.
3. Enable the sync plan, a new recurring logic should get created and associated to this sync plan.
_P.S: You will not be able to enable/disable a cancelled recurring logic._
***
1. Create a sync plan
2. Disable the sync plan and check state on recurring logic
3. Both should be disabled
4. Enable the sync plan.
5. Both should now be enabled.

### CODE REVIEW

File: app/views/katello/api/v2/sync_plans/show.json.rabl [hunk 1]
```diff
@@ -31,3 +31,8 @@ end
 node :sync_date do |sync_plan|
   sync_plan.plan_date_time
 end
+
+node :enabled do |sync_plan|
+    rec_logic_state = sync_plan.foreman_tasks_recurring_logic.state
+    (rec_logic_state == "active")? true : false
```
Comment (jlsherrill) [inline app/views/katello/api/v2/sync_plans/show.json.rabl]: Any reason to not just use:
```
sync_plan.foreman_tasks_recurring_logic.state == 'active' 
```

**Comment by jlsherrill** [comment]
I think we'd want to drop the 'enabled' column on sync plans right?

File: app/views/katello/api/v2/sync_plans/show.json.rabl (hunk 1 above)
Comment (sjha4) [reply app/views/katello/api/v2/sync_plans/show.json.rabl]: Yes..That is so much better! :+1:

File: app/views/katello/api/v2/sync_plans/show.json.rabl [hunk 2]
```diff
@@ -31,3 +31,7 @@ end
 node :sync_date do |sync_plan|
   sync_plan.plan_date_time
 end
+
+node :enabled do |sync_plan|
+  sync_plan.foreman_tasks_recurring_logic.state == 'active'
+end
```
Comment (sjha4) [inline app/views/katello/api/v2/sync_plans/show.json.rabl]: @jlsherrill : For the enabled flag on sync plan, we get the flag from rec logic directly..The value that API returns and what's stored on the field might be different..Do we want a task to correct that?

**Comment by sjha4** [comment]
Updated the PR to handle removing the enabled field from sync plan.

**Comment by sjha4** [comment]
Updates based on discussions with Justin offline:
This PR will be updated to contain only the migration to delete enabled field after the PR https://github.com/Katello/katello/pull/8008 is merged.

**Comment by sjha4** [comment]
Closing this.

**Pull request closed without merging**